src/databases/Curve3D/avtCurve3DFileFormat.C

```diff
@@ -0,0 +1,1122 @@
+// Copyright (c) Lawrence Livermore National Security, LLC and other VisIt
+// Project developers.  See the top-level LICENSE file for dates and other
+// details.  No copyright assignment is required to contribute to VisIt.
+
+// ************************************************************************* //
+//                            avtCurve3DFileFormat.C                         //
+// A reader for reading files that contain 3D curves. These may be named using
+// the same convention as in Curve2D, or not.
+// White space can be considered as a segmentation when MULTSEG is set.
+// In that case, we get the same behaviour as the one of gnuplot with the splot
+// function.
+// 2023/10/05 Olivier Cessenat.
+// Verbosity for poor man's debugging
+#define INTERACTIVEPLOT 0
+// Consider white space indicates a split
+#define MULTSEG 1
+// ************************************************************************* //
+
+#include <avtCurve3DFileFormat.h>
+
+#include <vector>
+#include <string>
+
+#include <vtkPointData.h>
+#include <vtkUnstructuredGrid.h>
+
+#include <vtkCellArray.h>
+#include <vtkPoints.h>
+#include <vtkCellType.h>
+
+#include <avtIntervalTree.h>
+#include <avtMaterial.h>
+
+#include <avtCallback.h>
+#include <avtDatabaseMetaData.h>
+
+#include <DebugStream.h>
+#include <DBOptionsAttributes.h>
+#include <StringHelpers.h>
+#include <InvalidFilesException.h>
+#include <InvalidVariableException.h>
+
+#include "visit_gzstream.h"
+
+#include <errno.h>
+#include <float.h>
+#include <stdlib.h>
+
+#ifdef PARALLEL
+#include <mpi.h>
+#include <avtParallel.h>
+#endif
+
+using std::vector;
+using std::string;
+using namespace std;
+
+// ****************************************************************************
+//  Method: avtCurve3DFileFormat constructor
+//
+//  Arguments:
+//      fname    The name of the curve file.
+//
+//  Programmer:  Hank Childs
+//  Creation:    May 28, 2002
+//
+//  Modifications:
+//
+//    Hank Childs, Tue Apr  8 11:09:03 PDT 2003
+//    Do not do so much work in the constructor.
+//
+//    Kathleen Bonnell, Fri Oct 28 13:02:51 PDT 2005
+//    Added curveTime, curveCycle.
+//
+//    Kathleen Biagas, Fri Aug 31 14:22:49 PDT 2018
+//    Added read options (currently unused).
```

Comment:
In all cases, please trim out all pre-existing modification comments and just include a remark along the lines of ...

```
Copied from Curve2D plugin as of <date> and adjusted for 3D
```

Reply:
OK